nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/test/java/org/apache/nifi/processors/elasticsearch/SearchElasticsearchTest.java

```diff
@@ -0,0 +1,217 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.elasticsearch;
+
+import org.apache.nifi.components.state.Scope;
+import org.apache.nifi.processors.elasticsearch.api.PaginationType;
+import org.apache.nifi.processors.elasticsearch.api.ResultOutputStrategy;
+import org.apache.nifi.state.MockStateManager;
+import org.apache.nifi.util.TestRunner;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.time.Instant;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class SearchElasticsearchTest extends AbstractPaginatedJsonQueryElasticsearchTest {
+    @BeforeAll
+    public static void setUpBeforeClass() throws Exception {
+        AbstractPaginatedJsonQueryElasticsearchTest.setUpBeforeClass();
+    }
+
+    public AbstractPaginatedJsonQueryElasticsearch getProcessor() {
+        return new SearchElasticsearch();
+    }
+
+    public boolean isStateUsed() {
+        return true;
+    }
+
+    public boolean isInput() {
+        return false;
+    }
+
+    @Test
+    public void testScrollError() {
+        final TestRunner runner = createRunner(false);
+        final TestElasticsearchClientService service = AbstractJsonQueryElasticsearchTest.getService(runner);
+        service.setMaxPages(2);
+        service.setThrowErrorInSearch(false);
+        runner.setProperty(AbstractPaginatedJsonQueryElasticsearch.PAGINATION_TYPE, PaginationType.SCROLL.getValue());
+        runner.setProperty(AbstractJsonQueryElasticsearch.QUERY, matchAllWithSortByMsgWithSizeQuery);
+
+        // initialize search
+        runOnce(runner);
+        AbstractJsonQueryElasticsearchTest.testCounts(runner, 0, 1, 0, 0);
+        runner.clearTransferState();
+
+        // scroll (error)
+        service.setThrowErrorInSearch(true);
+        runOnce(runner);
+        AbstractJsonQueryElasticsearchTest.testCounts(runner, 0, 0, 0, 0);
+        assertTrue(runner.getLogger().getErrorMessages().stream().anyMatch(logMessage ->
+                logMessage.getMsg().contains("Could not query documents") && logMessage.getThrowable().getMessage().contains("Simulated IOException - scroll")));
+    }
+
+    @Test
+    public void testScrollExpiration() throws Exception {
+        testPaginationExpiration(PaginationType.SCROLL);
+    }
+
+    @Test
+    public void testPitExpiration() throws Exception {
+        testPaginationExpiration(PaginationType.POINT_IN_TIME);
+    }
+
+    @Test
+    public void testSearchAfterExpiration() throws Exception {
```

Comment:
```suggestion
   void testScrollExpiration() throws Exception {
        testPaginationExpiration(PaginationType.SCROLL);
    }

    @Test
   void testPitExpiration() throws Exception {
        testPaginationExpiration(PaginationType.POINT_IN_TIME);
    }

    @Test
    void testSearchAfterExpiration() throws Exception {
```

No need for the `public` modifier on this test cases when using JUnit 5

Reply:
I realize there is no need for the `public`  modifier but I seemed to have this conversation with @exceptionfactory  in an earlier ticket (cannot find right now) whether to drop `public` or not and he was in favor of keeping it and having a follow on ticket which would handle removing the `public` modifier from all the unit tests. 
@exceptionfactory  please let me know whether you are still in favor of keeping the `public` modifier or can we remove it? Thanks!

Reply:
Personally I'd suggest just removing these here as you're already modifying the entire file/class anyway (and the warnings in IntelliJ are annoying, but one can live with those if there's a plan to wholesale change all `public` accessors in the future)

Reply:
@ChrisSamo632 Should I then remove public the unit tests in all the classes that I have refactored?

Reply:
@dan-s1 that's probably the approach I'd take, although I think these were the only methods I saw the warning for last night (but I didn't click through all the migrated classes, just the three where `testPagination` is being looked at)

but again, I'm happy to accept this staying as-is for now if there's a plan to remove them in the near future